File: app/models/quick_script/swars/tactic_stat_script/tactic_info.rb
class QuickScript::Swars::TacticStatScript
  class TacticInfo
    include ApplicationMemoryRecord
    memory_record [
      {
        key: :attack,
        name: "戦型",
      },
      {
        key: :defense,
        name: "囲い",
      },
      {
        key: :technique,
        name: "手筋",
      },
      {
        key: :note,
        name: "備考",
        # el_message: "異なるグループが混在しているため全体での比較には意味がない。たとえば居飛車と入玉を比べても意味がない。居飛車なら振り飛車と比べる。",
      },
    ]

    def ancestor_info
      Bioshogi::Explain::TacticInfo.fetch(key)
    end

    def tag_table
      "#{key}_tags"
    end
  end
end

File: nuxt_side/components/models/tactic_info.js
import { ApplicationMemoryRecord } from "@/components/models/application_memory_record.js"

export class TacticInfo extends ApplicationMemoryRecord {
  static get define() {
    return [
      { key: "attack",    name: "戦法", },
      { key: "defense",   name: "囲い", },
      { key: "technique", name: "手筋", },
      { key: "note",      name: "備考", },
    ]
  }

  get tag_list_name() {
    return `${this.key}_tag_list`
  }
}
